### PR TITLE
http timeout integration test: wait for 10ms for upstream reset

### LINKED ISSUE #6638: http_timeout_integration_test flake
See https://circleci.com/gh/envoyproxy/envoy/200799?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link.

### PR DESCRIPTION
**Pull request by mpuncel**
This test waits for the upstream to see a reset which confirms that the
router filter did the right thing when the global timeout is hit.
However since this involves the network, we would occasionally see the
reset after the wait call. Since we were waiting for 0ms we'd get
flakes. 10ms is hopefully high enough that the test will succeed
reliably.

Signed-off-by: Michael Puncel <mpuncel@squareup.com>

Description: address an integration test flake by sleeping longer when waiting for a remote reset to occur.
Risk Level: low
Testing: integration test 
Docs Changes: n/a 
Release Notes: n/a
Fixes #6638 
[Optional Deprecated:]

### CODE REVIEW

**Comment by mpuncel** [comment]
I'm not able to reproduce the flake locally after this change

File: test/integration/http_timeout_integration_test.cc [hunk 1]
```diff
@@ -36,7 +36,7 @@ TEST_P(HttpTimeoutIntegrationTest, GlobalTimeout) {
 
   // Ensure we got a timeout downstream and canceled the upstream request.
   response->waitForHeaders();
-  ASSERT_TRUE(upstream_request_->waitForReset(std::chrono::milliseconds(0)));
+  ASSERT_TRUE(upstream_request_->waitForReset(std::chrono::milliseconds(10)));
```
Comment (mattklein123) [inline test/integration/http_timeout_integration_test.cc]: 10ms is not very long on a loaded CI server. Why not just wait for a much longer period of time like 15s?

Comment (mpuncel) [reply test/integration/http_timeout_integration_test.cc]: sounds good to me, will change

**Pull request merged**